File: SDKs/googleapis.com/vision/v1p1beta1/python/src/sdk/models/shared/googlecloudvisionv1p1beta1page.py
import dataclasses
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import googlecloudvisionv1p1beta1block as shared_googlecloudvisionv1p1beta1block
from ..shared import googlecloudvisionv1p1beta1textannotationtextproperty as shared_googlecloudvisionv1p1beta1textannotationtextproperty


@dataclass_json
@dataclasses.dataclass
class GoogleCloudVisionV1p1beta1Page:
    r"""GoogleCloudVisionV1p1beta1Page
    Detected page from OCR.
    """
    
    blocks: Optional[list[shared_googlecloudvisionv1p1beta1block.GoogleCloudVisionV1p1beta1Block]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('blocks') }})
    confidence: Optional[float] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('confidence') }})
    height: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('height') }})
    property: Optional[shared_googlecloudvisionv1p1beta1textannotationtextproperty.GoogleCloudVisionV1p1beta1TextAnnotationTextProperty] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('property') }})
    width: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('width') }})
    

File: SDKs/googleapis.com/vision/v1p1beta1/java/lib/src/main/java/openapisdk/models/shared/GoogleCloudVisionV1p1beta1Page.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * GoogleCloudVisionV1p1beta1Page
 * Detected page from OCR.
**/
public class GoogleCloudVisionV1p1beta1Page {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("blocks")
    public GoogleCloudVisionV1p1beta1Block[] blocks;
    public GoogleCloudVisionV1p1beta1Page withBlocks(GoogleCloudVisionV1p1beta1Block[] blocks) {
        this.blocks = blocks;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("confidence")
    public Float confidence;
    public GoogleCloudVisionV1p1beta1Page withConfidence(Float confidence) {
        this.confidence = confidence;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("height")
    public Integer height;
    public GoogleCloudVisionV1p1beta1Page withHeight(Integer height) {
        this.height = height;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("property")
    public GoogleCloudVisionV1p1beta1TextAnnotationTextProperty property;
    public GoogleCloudVisionV1p1beta1Page withProperty(GoogleCloudVisionV1p1beta1TextAnnotationTextProperty property) {
        this.property = property;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("width")
    public Integer width;
    public GoogleCloudVisionV1p1beta1Page withWidth(Integer width) {
        this.width = width;
        return this;
    }
}